hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/CreateDagVisitor.java

```diff
@@ -215,13 +217,29 @@ public Vertex onNestedLoopJoin(JoinNestedLoopPhysicalRel rel) {
     }
 
     public Vertex onRoot(JetRootRel rootRel) {
-        Vertex vertex = dag.newUniqueVertex("ClientSink",
-                rootResultConsumerSink(rootRel.getInitiatorAddress()));
+        Vertex vertex;
+        RelNode input = rootRel.getInput();
+        if (input instanceof SortPhysicalRel) {
+            SortPhysicalRel limitRel = (SortPhysicalRel) input;
+            Expression<?> fetch = limitRel.fetch();
```

Comment:
We use only the `fetch` field of the `limitRel`. I think we should at least assert that offset is `0` and that there's no collation. It currently isn't possible because the unsupported-feature-verifier checks there's no ORDER BY, but who know where else Calcite uses the `Sort` node, it's better to assert it.